mass_mailing_partner/models/mail_mass_mailing_contact.py

```diff
@@ -14,68 +14,82 @@ class MailMassMailingContact(models.Model):
     partner_id = fields.Many2one(comodel_name='res.partner', string="Partner",
                                  domain=[('email', '!=', False)])
 
-    _sql_constraints = [
-        ('partner_list_uniq', 'unique(partner_id, list_id)',
-         _('Partner already exists in this mailing list.'))
-    ]
+    @api.constrains('partner_id', 'list_ids')
+    def _check_partner_id_list_ids(self):
+        for contact in self:
+            if contact.partner_id:
```

Comment:
I'm thinking that it's best to not limiting this, as it will lead to constant problems